include/sys/syscall_lookup.h

```diff
@@ -218,6 +218,11 @@ SYSCALL_LOOKUP(pwrite,                     4)
 #ifdef CONFIG_EVENT_FD
   SYSCALL_LOOKUP(eventfd,                  2)
 #endif
+#ifdef CONFIG_TIMER_FD
+  SYSCALL_LOOKUP(timerfd_create,           2)
+  SYSCALL_LOOKUP(timerfd_settime,          4)
+  SYSCALL_LOOKUP(timerfd_gettime,          2)
```

Comment:
it's better to implement timerfd_settime/timerfd_gettime through ioctl to reduce the syscall number.

Reply:
Could you please point me to some example? I just followed `eventfd` as an example.

Reply:
The idea is simple:

1. Add ioctl(e.g. TIMERFD_SETTIME, TIMER_GETTIME)
2. Move the code from timerfd_settime/timerfd_getime) to timerfd_ioctl
3. Implement timerfd_settime and timerfd_gettime in libc as the simple ioctl wrapper.

Reply:
I see no problems with `timerfd_gettime` in such approach, but not sure about `timerfd_settime` that needs 3 arguments to be passed. In general `int flags, const struct itimerspec *new_value, struct itimerspec *old_value` should be packed into intermediate structure and pointer to that structure has to be passed to `ioctl` call. I also do not have any issues with that, but the question comes where the declaration of that intermediate type should be located? Should I call something like
```
typedef struct
{
  int flags;
  const struct itimerspec *new_value;
  struct itimerspec *old_value;
} timerfd_settime_t
```
into `timerfd.h`?

Reply:
Also `eventfd_read` and `eventfd_write` are `glibc` functions by definition. While `timerfd` APIs are a syscalls in Linux

Reply:
> I see no problems with `timerfd_gettime` in such approach, but not sure about `timerfd_settime` that needs 3 arguments to be passed. In general `int flags, const struct itimerspec *new_value, struct itimerspec *old_value` should be packed into intermediate structure and pointer to that structure has to be passed to `ioctl` call. I also do not have any issues with that, but the question comes where the declaration of that intermediate type should be located? Should I call something like
> 
> ```
> typedef struct
> {
>   int flags;
>   const struct itimerspec *new_value;
>   struct itimerspec *old_value;
> } timerfd_settime_t;
> ```
> 
> into `timerfd.h`? @xiaoxiang781216 what is your opinion?

Yes, it's reasonable. BTW, it's better to use struct directly and change pointer to value, like this:
```
 struct timerfd_settime_s
 {
   int flags;
   struct itimerspec new_value;
   struct itimerspec old_value;
 };
```

> Also `eventfd_read` and `eventfd_write` are `glibc` functions by definition. While `timerfd` APIs are a syscalls in Linux

Once we move the action into `timerfd_ioctl`, `timerfd_settime` and `timerfd_gettime` can be changed to 'libc' function like `eventfd_read` and `eventfd_write`. Only, timerfd_create need be syscall.

Reply:
Can this be done in 2 steps? I really do not mind to go this way, but since such change goes in another path compared to Linux and require additional retesting and significant rework (at least IMO) I would like to open an enhancement issue and involve some more people into discussion instead of doing changes in this  PR. @xiaoxiang781216 does this sounds reasonable?

Reply:
Sure, let's do in the next step.

> BTW going with
> 
> ```
> struct timerfd_settime_s {
>  int flags;
>  struct itimerspec new_value;
>  struct itimerspec old_value;
> };
> ```
> 
> is not the best as `old_value` is optional parameter that can be `NULL`.

timerfd_ioctl could return old_value always, but timerfd_settime can skip the assignment in case of old_value is NULL.